### PR TITLE
Implement new shares calculation mechanism

### LINKED ISSUE #10: Improve initial shares calculation algorithm

### PR DESCRIPTION
**Pull request by iorveth**
Implements initial shares calculation mechanism as described 

https://uniswap.org/whitepaper.pdf 
3.4 Initialization of liquidity token supply

Closes https://github.com/subdarkdex/subdex-parachain/issues/10

### CODE REVIEW

File: pallets/dex-pallet/src/exchange.rs [hunk 1]
```diff
@@ -23,20 +23,52 @@ impl<T: Trait> Default for Exchange<T> {
...
         sender: T::AccountId,
     ) -> Result<Self, Error<T>> {
         let mut shares_map = BTreeMap::new();
-        shares_map.insert(sender, T::InitialShares::get());
+        let min_fee = Self::get_min_fee();
+
+        let initial_shares = Self::sqrt(first_asset_amount * second_asset_amount)
+            .checked_sub(&min_fee)
+            .ok_or(Error::<T>::UnderflowOccured)?;
+
```
Comment (whalelephant) [inline pallets/dex-pallet/src/exchange.rs:63]: would zero be ok here, after `checked_sub`?

File: pallets/dex-pallet/src/exchange.rs [hunk 2]
```diff
@@ -23,20 +23,52 @@ impl<T: Trait> Default for Exchange<T> {
 }
 
 impl<T: Trait> Exchange<T> {
+    // Avoid casting to float
+    fn sqrt(y: BalanceOf<T>) -> BalanceOf<T> {
```
Comment (whalelephant) [inline pallets/dex-pallet/src/exchange.rs:27]: can we limit max value here?

File: pallets/dex-pallet/src/exchange.rs (hunk 1 above)
Comment (iorveth) [reply pallets/dex-pallet/src/exchange.rs:63]: Hmm, let`s comment `checked_sub` part for now, i hope, it should be fairly safe even without having this.

File: pallets/dex-pallet/src/exchange.rs (hunk 2 above)
Comment (iorveth) [reply pallets/dex-pallet/src/exchange.rs:27]: We shouldn`t have an overflow issue here.

Comment (iorveth) [reply pallets/dex-pallet/src/exchange.rs:27]: I implemented an overflow/underflow check here, though

**Pull request merged**